File: Application/Services/Operations/StockProduct/Dtos/ProductKind/ProductTypeDto.cs
using System.Collections.Generic;
using Application.Services.Shared.Dtos;

namespace Application.Services.Operations.StockProduct.ProductKind
{

    public class ProductTypeDto: RootBaseDto
    {
        public string Name { get; set; }
        public List<SegmentDto> Segments { get; set; }
         public List<ProductDto> Products { get; set; }
        
        
    }
}

File: Views/src/components/stock-product/product/dtos/product-type-dto.ts
import { RootBase } from "src/shared/entities-dtos/root-base";
import { SegmentDto } from "./segment-dto";
import { ProductDto } from "./product-dto";

export class ProductTypeDto extends RootBase {
    name: string;
    segments: SegmentDto[];
    products: ProductDto[];
}